### PR TITLE
User login

### PR DESCRIPTION
**Pull request by icerman77**
Pivotal Tracker user story
https://www.pivotaltracker.com/story/show/171561350 
```
As a user
In order to access my account
I would like to be able log in
```
## Changes proposed in this pull request:
- Making the user login/authentication function.
## What we learn working on the feature
- Using mocks in the frontend to get a successfully login we learn while working on this user story.

### CODE REVIEW

File: cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
+  "body": "Fixtures are a great way to mock data for responses to routes"
+}
```
Comment (tochman) [inline cypress/fixtures/example.json]: Why did you keep this fixture around?

File: cypress/fixtures/profile.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "id": 8739,
+  "name": "Jane",
+  "email": "jane@example.com"
+}
```
Comment (tochman) [inline cypress/fixtures/profile.json]: What is this fixture used for?

File: cypress/fixtures/users.json [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    },
+    "phone": "024-648-3804",
+    "website": "ambrose.net",
+    "company": {
+      "name": "Hoeger LLC",
+      "catchPhrase": "Centralized empowering task-force",
+      "bs": "target end-to-end models"
+    }
+  }
+]
```
Comment (tochman) [inline cypress/fixtures/users.json]: What is this fixture used for?

File: cypress/integration/userCanLogin.feature.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      cy.visit("/");
+  });
+  it("successfully", () => {
+      cy.route({
+          method: "POST",
+          url: "http://localhost:3000/api/v1/auth/sign_in",
+          response: "fixture:login.json",
+          headers: {
+              uid: "user@mail.com"
+          }
```
Comment (tochman) [inline cypress/integration/userCanLogin.feature.js]: Why do you include this header in the mocked response?

File: cypress/integration/userCanLogin.feature.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          response: "fixture:login.json",
+      });
+      cy.get("#login").click();
+      cy.get("#login-form").within(() => {
+          cy.get("#email").type("user@mail.com");
+          cy.get("#password").type("password");
+          cy.get('button').contains('Submit').click()
+      });
+      cy.get("#message").should("contain", "Welcome back");
+  });
```
Comment (faraznaeem) [inline cypress/integration/userCanLogin.feature.js:21]: code formatting

File: src/App.jsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import LoginForm from "./Components/LoginForm";
+import { authenticate } from './modules/auth';
+
+class App extends Component {
+  state = {
+    renderLoginForm: false,
+    renderLoginForm: false,
+    authenticated: false,
+    message: ""
+
```
Comment (faraznaeem) [inline src/App.jsx]: formatting

**Review by faraznaeem — CHANGES_REQUESTED** [review]
Make sure that you format your code properly before submission

File: cypress/fixtures/example.json (hunk 1 above)
Comment (icerman77) [reply cypress/fixtures/example.json]: We forgot to delete this when we were deleting extra files that cypress creates.

File: cypress/fixtures/profile.json (hunk 2 above)
Comment (icerman77) [reply cypress/fixtures/profile.json]: We forgot to delete this when we were deleting extra files that cypress creates.

File: cypress/fixtures/users.json (hunk 3 above)
Comment (icerman77) [reply cypress/fixtures/users.json]: We forgot to delete this when we were deleting extra files that cypress creates.

File: cypress/integration/userCanLogin.feature.js (hunk 4 above)
Comment (icerman77) [reply cypress/integration/userCanLogin.feature.js]: We forgot to delete this when we were deleting extra files that cypress creates.

File: src/App.jsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
+import LoginForm from "./Components/LoginForm";
+import { authenticate } from './modules/auth';
+
+class App extends Component {
+  state = {
+    renderLoginForm: false,
```
Comment (oliverochman) [inline src/App.jsx:7]: duplicates

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: cypress/integration/userCanLogin.feature.js (hunk 5 above)
Comment (icerman77) [reply cypress/integration/userCanLogin.feature.js:21]: Fixed the formatting

File: src/App.jsx (hunk 6 above)
Comment (icerman77) [reply src/App.jsx]: Fixed the formatting

File: src/App.jsx (hunk 7 above)
Comment (icerman77) [reply src/App.jsx:7]: Deleted the duplicated line.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**